homeassistant/components/binary_sensor/egardia.py

```diff
@@ -0,0 +1,98 @@
+"""
+Interfaces with Egardia/Woonveilig alarm control panel.
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/binary_sensor.egardia/
+"""
+import logging
+import asyncio
+
+from homeassistant.components.binary_sensor import BinarySensorDevice
+from homeassistant.const import (STATE_ON, STATE_OFF)
+_LOGGER = logging.getLogger(__name__)
+ATTR_DISCOVER_DEVICES = 'egardia_sensor'
+D_EGARDIASYS = 'egardiadevice'
+
+EGARDIA_TYPE_TO_DEVICE_CLASS = {'IR Sensor': 'motion',
+                                'Door Contact': 'opening',
+                                'IR': 'motion'}
+
+
+def _get_device_class(egardia_type):
+    return EGARDIA_TYPE_TO_DEVICE_CLASS.get(egardia_type, None)
+
+
+def _get_sensor_state(egardia_input):
+    if egardia_input:
+        return STATE_ON
+    else:
+        return STATE_OFF
+
+
+def _create_sensor(hass, sensor):
+    return EgardiaBinarySensor(
+        hass, senid=sensor["id"],
+        name=sensor['name'],
+        state=_get_sensor_state(sensor['cond']),
+        device_class=_get_device_class(sensor['type'])
+        )
+
+
+@asyncio.coroutine
+def async_setup_platform(hass, config, async_add_devices, discovery_info=None):
+    """Initialize the platform."""
+    if (discovery_info is None or
+            discovery_info[ATTR_DISCOVER_DEVICES] is None):
+        return
+
+    discinfo = discovery_info[ATTR_DISCOVER_DEVICES]
+
+    # multiple devices here!
+    async_add_devices(
+        _create_sensor(hass, discinfo[sensor])
+        for sensor in discinfo
+        )
```

Comment:
Add True as second argument so that update is called after adding the device.

Reply:
As second argument to what? `async_add_devices`? But how?

Reply:
Yes.
```py
async_add_devices(..., True)
```

Reply:
Ok, thanks. Done.